receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -234,9 +234,23 @@ func addGaugeDatapoints(rm pmetric.ResourceMetrics, ls labels.Labels, ts writev2
 	// In OTel name+type+unit is the unique identifier of a metric and we should not create
 	// a new metric if it already exists.
 
-	// TODO: Check if Scope is already present by comparing labels "otel_scope_name" and "otel_scope_version"
-	// with Scope.Name and Scope.Version. If it is present, we should append to the existing Scope.
-	m := rm.ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetEmptyGauge()
+	scopeName := ls.Get("otel_scope_name")
+	scopeVersion := ls.Get("otel_scope_version")
+	// Check if the name and version present in the labels are already present in the ResourceMetrics.
+	// If it is not present, we should create a new ScopeMetrics.
+	// Otherwise, we should append to the existing ScopeMetrics.
+	for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+		scope := rm.ScopeMetrics().At(j)
+		if scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
```

Comment:
Is there a possibility that `scopeName` and `scopeVersion` could be empty strings? If so, should we implement different behavior to handle that case?

Reply:
Hmmm, I'm not sure if it's possible to receive labels with empty strings... but if they are I believe we treat them as "never match"

Reply:
```suggestion
		if scopeName != "" && scopeVersion != "" && scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
```

I think that's all...?

Reply:
I think the current implementation is correct.  We don't want to do `scope := rm.ScopeMetrics().AppendEmpty()` for each metric with an empty `otel_scope_name`.